specification/imds/data-plane/Microsoft.InstanceMetadataService/stable/2019-08-15/imds.json

```diff
@@ -0,0 +1,975 @@
+                    } 
+                ],
+                "responses": {
+                    "200": {
+                        "description": "OK",
+                        "schema": {
+                            "$ref": "#/definitions/AttestedData"
+                        }
+                    },
+                    "default": {
+                        "description": "Error response describing why the operation failed.",
+                        "schema": {
+                            "$ref": "#/definitions/ErrorResponse"
+                        }
+                    }
+                },
+                "x-ms-examples": {
+                    "Get Attested Data for the VM": {
+                        "$ref": "./examples/GetAttestedData.json"
+                    }
+                }
+            }
+        },
+        "/identity/oauth2/token": {
+            "get": {
+                "tags": [
+                    "Get token"
+                ],
+                "description": "Get a Token from Azure AD",
+                "operationId": "Identity_GetToken",
+                "produces": [
+                    "application/json"
+                ],
+                "security": [
+                    {
+                    },
+                    {
+                        "basic_auth": []
+                    }
+                ],
+                "parameters": [
+                    {
+                        "$ref": "#/parameters/MetadataParameter"
+                    },
+                    {
+                        "name": "resource",
+                        "in": "query",
+                        "type": "string",
+                        "description": "This is the urlencoded identifier URI of the sink resource for the requested Azure AD token. The resulting token contains the corresponding aud for this resource.",
+                        "required": true
+                    },
+                    {
+                        "$ref": "#/parameters/ApiVersionParameter"
+                    },
+                    {
+                        "name": "client_id",
+                        "in": "query",
+                        "type": "string",
+                        "description": "This identifies, by Azure AD client id, a specific explicit identity to use when authenticating to Azure AD. Mutually exclusive with object_id and msi_res_id.",
+                        "required": false
+                    },
+                    {
+                        "name": "object_id",
+                        "in": "query",
+                        "type": "string",
+                        "description": "This identifies, by Azure AD object id, a specific explicit identity to use when authenticating to Azure AD. Mutually exclusive with client_id and msi_res_id.",
+                        "required": false
+                    },
+                    {
+                        "name": "mi_res_id",
```

Comment:
Changing the name of the parameter is a breaking change (it was `msi_res_id` in the previous version). Why is this change necessary?

Reply:
The name change was based on code review feedback from the Identity team. But I leave it out for now I changed back to `msi_res_id`.

Reply:
What does the API actually expect? Was the description incorrect earlier and this was a correction, or was this a planned change to the REST API?

Reply:
I am not 100% sure if imds is responding mi_res_id today.  So I'd rather leave this particular change to the Identity team.

Reply:
This needs to be sorted out before we can merge the PR. The description has to be correct w. regards to the actual REST API.

Reply:
@johanste , do you mean whether `msi_res_id` or `mi_res_id` should be sorted it now? I looked file history, it has been `msi_res_id`. So no change for this version.

Reply:
@johanste I looked identity team's source code, there are no changes in the history about msi_res_id. Also public docs are about msi_res_id. 
This version is particularly for our hybrid scenario. No impact to the existing Azure IMDS. 
If you have concerns, I can follow-up with the Metadata and Identity teams about it. However to fix the existing spec issues, can we  make a separate PR to fix the existing api versions all together. Thoughts?

Reply:
Assuming that the name in the swagger is what the REST API actually expects/uses, then we are good. From you explanation, that seems to be the case. What we should *not* do is check in a swagger that is incorrect and then later fix it. 

If the service team intends to change the name in the REST API in the future, please make sure that they consult the azure api review board to determine if the change is warranted.

Reply:
Agreed.  teams should consult with the board first for any breaking changes. I will follow-up with the Metadata and Identity team to see if there are any inconsistencies between code and the spec.
For this particular PR, do you have any other concerns?